datadog_checks_downloader/datadog_checks/downloader/download.py

```diff
@@ -133,6 +139,10 @@ def __download_custom(self, target, extension):
 
         fileinfo = target.get('fileinfo', {})
         custom = fileinfo.get('custom', {})
+
+        if custom.get('root-layout-type', DEFAULT_ROOT_LAYOUT_TYPE) != self.root_layout_type:
+            return target_abspaths
```

Comment:
Hmm, not sure this is the right behaviour... what would this ultimately cause?

Reply:
Should we error instead?

Reply:
Yes, I think that is the much safer thing to do. We should create a new error type, bail by throwing that error, and inform the user accordingly.

Reply:
Please use the `.exceptions` module to add the new error